.github/workflows/main.yml

```diff
@@ -173,41 +173,34 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-22.04, macos-12, windows-2022]
+        os: [ubuntu-latest, macos-12, windows-latest]
```

Comment:
Same here, I remember wanting to keep on the latest images (this isn't the case for Mac, not sure why). So the other adapter repos use `latest` instead of a hard coded version so updating this to keep parity as well. I can revert this if it isn't wanted

Reply:
We determined that it's better to hard pin and then upgrade as we can. The primary issue is that the latest tag eventually bumps and the new image may not have all of the python versions installed. We ran into this on macos in particular, which is probably what we used to pin that in the past.